clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3268,12 +3268,29 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &AL) {
 
   ASTContext &Ctx = S.getASTContext();
 
+  // The arguments to reqd_work_group_size are ordered based on which index
+  // increments the fastest. In OpenCL, the first argument is the index that
+  // increments the fastest, and in SYCL, the last argument is the index that
+  // increments the fastest.
+  //
+  // [[sycl::reqd_work_group_size]] and [[cl::reqd_work_group_size]] are
+  // available in SYCL modes and follow the SYCL rules.
+  // __attribute__((reqd_work_group_size)) is only available in OpenCL mode
+  // and follows the OpenCL rules.
   if (const auto *A = D->getAttr<SYCLIntelMaxWorkGroupSizeAttr>()) {
-    if (!((getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getXDimVal()) &&
-          (getExprValue(AL.getArgAsExpr(1), Ctx) <= *A->getYDimVal()) &&
-          (getExprValue(AL.getArgAsExpr(2), Ctx) <= *A->getZDimVal()))) {
+    bool checkFirstArgument = S.getLangOpts().OpenCL
+              ? getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getZDimVal()
+	      : getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getXDimVal();
+    bool checkSecondArgument =
+	      getExprValue(AL.getArgAsExpr(1), Ctx) <= *A->getYDimVal();
+    bool checkThirdArgument = S.getLangOpts().OpenCL
```

Comment:
```suggestion
    bool CheckFirstArgument = S.getLangOpts().OpenCL
              ? getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getZDimVal()
	      : getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getXDimVal();
    bool CheckSecondArgument =
	      getExprValue(AL.getArgAsExpr(1), Ctx) <= *A->getYDimVal();
    bool CheckThirdArgument = S.getLangOpts().OpenCL
```
Coding style guideline nits.